File: ios/RNZendeskUser.swift
import ZendeskSDK

@objc(RNZendeskUser)
class RNZendeskUser: NSObject {
    var id: String
    var externalId: String

  init(_ user: ZendeskUser) {
      self.id = user.id
      self.externalId = user.externalId
  }

  func asDictionary() -> NSDictionary {
    return [
      "id" : self.id,
      "externalId" : self.externalId,
    ]
  }
}

File: android/src/main/java/com/reactnativezendeskmessaging/RNZendeskUser.java
public class RNZendeskUser {
  public String id;
  public String externalId;

  public RNZendeskUser(ZendeskUser user) {
    this.id = user.getId();
    this.externalId = user.externalId();
  }

  public ReadableMap<String, String> asReadableMap() {
    return new ReadableMap<String, String>() {{
      put("id", this.id);
      put("externalId", this.externalId);
    }};
  }
}
